src/test/java/com/volasoftware/tinder/services/implementations/FriendServiceImplTest.java

```diff
@@ -254,14 +269,79 @@ public void testLinkAllAccountsWithBotsWhenRealAccountsDoesNotHaveBotsThenSucces
         when(accountRepository.save(realAccount2)).thenReturn(savedAccount2);
 
         //then
-        String result = friendService.linkAllAccountsWithBots(pageable);
+        String result = friendService.linkFriends(null, pageable);
         assertEquals(OperationConstant.SUCCESSFUL, result);
     }
 
-    private Account createAccount(Long id, AccountType type) {
+
+    @Test
+    public void testSortFriendsByLocation() {
+        //given
+        FriendSearchDto friendSearchDto = new FriendSearchDto();
+        friendSearchDto.setLatitude(0.0);
+        friendSearchDto.setLongitude(0.0);
+
+        List<FriendDto> expectedFriends = Arrays.asList(
+            FriendDto.builder().firstName("John").lastName("Doe")
+                .gender(Gender.MALE).age(25)
+                .locationDto(LocationDto.builder().latitude(10.0).longitude(10.0).build()).build(),
+            FriendDto.builder().firstName("Jane").lastName("Doe")
+                .gender(Gender.FEMALE).age(22)
+                .locationDto(LocationDto.builder().latitude(20.0).longitude(20.0).build()).build(),
+            FriendDto.builder().firstName("Alice").lastName("Doe")
+                .gender(Gender.OTHER).age(30)
+                .locationDto(LocationDto.builder().latitude(30.0).longitude(30.0).build()).build()
+        );
+
+        //when
+        List<Account> friendsList = Arrays.asList(friend1, friend2, friend3);
+        ArrayList<Account> friends = new ArrayList<>(friendsList);
+        when(accountRepository.findOneByEmail(any())).thenReturn(
+            Optional.ofNullable(loggedAccount));
+        when(friendMapper.accountListToFriendDtoList(friends)).thenReturn(expectedFriends);
+
+        List<FriendDto> expectedFriendDtos = friendMapper.accountListToFriendDtoList(friends);
```

Comment:
You are mocking the mapper method, but on line 303 you're using it. Is it possible the test to work without mapper mock (of course with mocked findFriendsByLocation() added)?